libnd4j/pom.xml

```diff
@@ -173,16 +175,19 @@
                 <executions>
                     <execution>
                         <id>libnd4j.test.run</id>
-                        <phase>test</phase>
+                        <!--FIXME: workaround to enable libnd4j tests-->
+                        <!--FIXME: To run tests before build process-resources was chosen-->
+                        <phase>${libnd4j.tests.phase}</phase>
                         <goals>
                             <goal>exec</goal>
                         </goals>
                         <configuration>
-                            <skip>${maven.test.skip}</skip>
+                            <!--<skip>${maven.test.skip}</skip>-->
```

Comment:
Let's put this back though?

Reply:
Can't do that, because idea is to run test with `mvn install -Dmaven.skip.test`. Which means that for the rest components test will be disabled, but for libnd4j not.

Reply:
We can run `mvn install -pl libnd4j` for that no?

Reply:
In general - yes, but i will be hard to track in our case...

Reply:
Anyway, it doesn't matter how it's achieved, but we need to have at least one flag like `skipTests` that allows up to skip all tests, and probably through a profile, so we can have a more complex activation conditions when we need. Could you work on that? That would be a fix for issue #5337

Reply:
Yeah, that is way initially I did this via profile... Should I revert it back?

Reply:
Sure, but it didn't look like it was doing that..

Reply:
Ok, drop last commit, which bring us to profile impl... This will allows us to have libnd4j tests skipping option, and tests will be run with the whole stack build.
At the same time temporary workaround to run tests during the install phase will also be there with the help of profile...

Reply:
Ok, now let's make it run by default unless `skipTests` or `maven.test.skip` is set. I think it would work with a profile like this as @AlexDBlack pointed out: https://github.com/deeplearning4j/deeplearning4j/blob/master/nd4j/pom.xml#L69-L91